content/zh/docs/concepts/extend-kubernetes/api-extension/custom_resources.md

```diff
@@ -0,0 +1,524 @@
+<!--
+* There is an existing, well-documented config file format, such as a `mysql.cnf` or `pom.xml`.
+* You want to put the entire config file into one key of a configMap.
+* The main use of the config file is for a program running in a Pod on your cluster to consume the file to configure itself.
+* Consumers of the file prefer to consume via file in a Pod or environment variable in a pod, rather than the Kubernetes API.
+* You want to perform rolling updates via Deployment, etc, when the file is updated.
+-->
+*存在一种记录良好的配置文件格式，，例如“mysql.cnf”或者“pom.xml”。
+*你想把所有的配置文件都放进配置映射的一个键中。
+*配置文件的作用就是，让运行在集群中pod的程序配置自身环境。
+*文件使用者更倾向于使用通过Pod中的文件或Pod中的环境变量，而不是使用Kubernetes API。
+*你希望在更新文件等时候，通过“部署”等执行滚动更新。
+<!--
+{{< note >}}
+Use a [secret](/docs/concepts/configuration/secret/) for sensitive data, which is similar to a configMap but more secure.
+{{< /note >}}
+-->
+{{< note >}}
+对敏感数据使用机密，该机密类似于配置映射，但更安全。
+{{</ note >}}
+<!--
+Use a custom resource (CRD or Aggregated API) if most of the following apply:
+-->
+如果以下大多数情况出现，请使用自定义资源：
+<!--
+* You want to use Kubernetes client libraries and CLIs to create and update the new resource.
+* You want top-level support from kubectl (for example: `kubectl get my-object object-name`).
+* You want to build new automation that watches for updates on the new object, and then CRUD other objects, or vice versa.
+* You want to write automation that handles updates to the object.
+* You want to use Kubernetes API conventions like `.spec`, `.status`, and `.metadata`.
+* You want the object to be an abstraction over a collection of controlled resources, or a summarization of other resources.
+-->
+*您希望使用 Kubernetes 客户端库和 CLIs 来创建和更新新资源。
+*你希望从kubectl得到顶级支持（比如：kubectl get my-object object-name）。
+*你希望新建一个自动化以监视新对象的更新，然后监视CRUD其他对象，反之亦然。
+*你希望编写处理对象更新的自动化。
+*你想要使用Kubernetes API约定，比如像.spec，.status，和.metadata。
+*你希望对象是受控资源集合的抽象，或者是其他资源的汇总。
+<!--
+ ## Adding custom resources
+-->
+添加自定义资源
+<!--
+Kubernetes provides two ways to add custom resources to your cluster:
+-->
+Kubernetes提供了两种将自定义资源添加到集群的方法：
+<!--
+- CRDs are simple and can be created without any programming.
+- [API Aggregation](/docs/concepts/extend-kubernetes/api-extension/apiserver-aggregation/) requires programming, but allows more control over API behaviors like how data is stored and conversion between API versions.
+-->
+-CRDs非常简单，无需任何编程即可创建。
+-API集合需要编程，但允许对API行为进行更多控制，比如数据的存储方式和API版本之间的转换。
+<!--
+Kubernetes provides these two options to meet the needs of different users, so that neither ease of use nor flexibility is compromised.
+-->
+Kubernetes提供了两个选项来满足不同使用者的需要，因此无论是易用性还是灵活性都不受影响。
+<!--
+Aggregated APIs are subordinate APIServers that sit behind the primary API server, which acts as a proxy. This arrangement is called [API Aggregation](/docs/concepts/extend-kubernetes/api-extension/apiserver-aggregation/) (AA). To users, it simply appears that the Kubernetes API is extended.
+-->
+API集合是位于主API服务器后面的从属API服务器，用来充当代理。这种安排被称作API集合（AA）。对于用户来说，Kubernetes API似乎只是扩展了。
```

Comment:
The hyperlink is missing. Also, I don't think it's necessary to retain “AA”.

Reply:
I've updated the file already. You can check it.